src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,231 @@
+const RIGHT_START: Directions = 'right-start'
+const RIGHT: Directions = 'right'
+const RIGHT_END: Directions = 'right-end'
+const BOTTOM_START: Directions = 'bottom-start'
+const BOTTOM: Directions = 'bottom'
+const BOTTOM_END: Directions = 'bottom-end'
+const LEFT_START: Directions = 'left-start'
+const LEFT: Directions = 'left'
+const LEFT_END: Directions = 'left-end'
 
-export const opposites = {
+
+export const opposites: Record<Directions, Directions> = {
+  [TOP_START]: BOTTOM_END,
   [TOP]: BOTTOM,
+  [TOP_END]: BOTTOM_START,
+  [BOTTOM_START]: TOP_END,
   [BOTTOM]: TOP,
+  [BOTTOM_END]: TOP_START,
+  [LEFT_START]: RIGHT_END,
   [LEFT]: RIGHT,
+  [LEFT_END]: RIGHT_START,
+  [RIGHT_START]: LEFT_END,
   [RIGHT]: LEFT,
-};
-export const arrowLabels = {
-  [TOP]: "↑",
-  [BOTTOM]: "↓",
-  [LEFT]: "←",
-  [RIGHT]: "→",
-};
-export const directions = [TOP, BOTTOM, LEFT, RIGHT];
-export const rotation = {
+  [RIGHT_END]: LEFT_START,
+}
+
+export const arrowLabels: Record<Directions, string> = {
+  [TOP_START]: '↑',
+  [TOP]: '↑',
+  [TOP_END]: '↑',
+  [BOTTOM_START]: '↓',
+  [BOTTOM]: '↓',
+  [BOTTOM_END]: '↓',
+  [LEFT_START]: '←',
+  [LEFT]: '←',
+  [LEFT_END]: '←',
+  [RIGHT_START]: '→',
+  [RIGHT]: '→',
+  [RIGHT_END]: '→',
+}
+export const directions: Directions[] = [
+  TOP_START,
+  TOP,
+  TOP_END,
+  BOTTOM_START,
+  BOTTOM,
+  BOTTOM_END,
+  LEFT_START,
+  LEFT,
+  LEFT_END,
+  RIGHT_START,
+  RIGHT,
+  RIGHT_END,
+]
+export const rotation: Record<Directions, number> = {
+  [LEFT_START]: -45,
   [LEFT]: -45,
+  [LEFT_END]: -45,
+  [TOP_START]: 45,
   [TOP]: 45,
+  [TOP_END]: 45,
+  [RIGHT_START]: 135,
   [RIGHT]: 135,
+  [RIGHT_END]: 135,
+  [BOTTOM_START]: -135,
   [BOTTOM]: -135,
-};
+  [BOTTOM_END]: -135,
+}
 
 export type AttentionState = {
-  isShowing?: boolean;
-  isCallout?: boolean;
-  actualDirection?: Directions;
-  directionName: Directions;
-  arrowEl?: HTMLElement | null;
-  attentionEl?: HTMLElement | null;
-  targetEl?: unknown;
-  top?: Boolean;
-  right?: Boolean;
-  bottom?: Boolean;
-  left?: Boolean;
-  tooltip?: Boolean;
-  popover?: Boolean;
-  callout?: Boolean;
-  noArrow?: Boolean;
-  waitForDOM?: () => void;
-};
+  isShowing?: boolean
+  isCallout?: boolean
+  actualDirection?: Directions
+  directionName?: Directions
+  arrowEl?: HTMLElement | null
+  attentionEl?: HTMLElement | null
+  flip?: Boolean
+  fallbackPlacements?: Directions[]
+  targetEl?: ReferenceElement | null
+  noArrow?: Boolean
+  distance?: number
+  skidding?: number
+  waitForDOM?: () => void
+}
+
+const middlePosition: string = 'calc(50% - 7px)'
+const isDirectionVertical = (name: Directions) =>
+  ([TOP_START, TOP, TOP_END, BOTTOM_START, BOTTOM, BOTTOM_END] as Directions[]).includes(name)
 
-const middlePosition = "calc(50% - 7px)";
-const isDirectionVertical = (name: string) => [TOP, BOTTOM].includes(name);
 function computeCalloutArrow({
   actualDirection,
-  directionName,
+  directionName = 'bottom',
```

Comment:
Maybe refer to this from the const above that's a type of `Directions`. This currently looks like a string without checking the type

Reply:
But isn't it already set as type Directions here? We are setting `actualDirection`, `directionName` and `arrowEl` to be part of the `AttentionState`. When I hover on `directionName` it says that is already of type `Directions`:

<img width="339" alt="Skärmavbild 2024-03-04 kl  09 57 00" src="https://github.com/warp-ds/core/assets/144792260/2cdc0391-f3e9-455b-9a16-ea339121e3eb">

 and if I try and replace `bottom` with just a random string, it complains saying that it is not part of the `Directions` type: 
<img width="420" alt="Skärmavbild 2024-03-04 kl  09 57 09" src="https://github.com/warp-ds/core/assets/144792260/851c17e8-8b2a-4079-ade2-c7561900f594">

Reply:
Yes it does but without hovering, it looks like a string. Since we already have a typed const in the file, wouldn't it be better to use that instead of re-defining a string?

Reply:
Which typed const in the file are you referring to, that I should use instead of re-defining a string? 🙈

Reply:
I could create a `defaultDirectionName` const and use that instead:   `const defaultDirectionName: Directions = 'bottom'`?

Reply:
Like this: [992972f](https://github.com/warp-ds/core/pull/15/commits/992972fa4d611cab12c3ac9e62d0645e36009a34)

Reply:
Updated to use `BOTTOM` enum instead of `defaultDirectionName` in this commit: [5056e3b](https://github.com/warp-ds/core/pull/15/commits/5056e3b31087983ad8bc9ff71b4259d5bbad0838)